network-core/src/lib.rs

```diff
@@ -10,6 +10,36 @@ pub mod error;
 pub mod client;
 pub mod server;
 
+pub mod gossip {
```

Comment:
That's more like it, though I don't like jamming stuff into `lib.rs`. Copy it out to a separate file, perhaps.
We may also want bikeshed the multiple modules named `gossip` at various levels. How about naming this module `types` and renaming `Error` to `NodeIdError`? Also try making it private, it should not be needed outside the crate.